src/create.cpp

```diff
@@ -144,19 +146,30 @@ create_file(std::string_view label, std::string_view extension,
                  nullptr, CREATE_NEW, FILE_ATTRIBUTE_NORMAL, nullptr);
   if (handle == INVALID_HANDLE_VALUE) {
     ec = std::error_code(GetLastError(), std::system_category());
-    return std::pair<fs::path, file::native_handle_type>();
+    return {};
   }
 #else
   // FIXME: `mkstemps` function is not a part of POSIX standard
   int handle = mkstemps(path.data(), static_cast<int>(extension.size()));
   if (handle == -1) {
     ec = std::error_code(errno, std::system_category());
-    return std::pair<fs::path, file::native_handle_type>();
+    return {};
   }
 #endif
 
+  std::ios::openmode mode = binary ? std::ios::binary : std::ios::openmode();
+  mode |= std::ios::in | std::ios::out;
+
+  std::fstream stream = std::fstream(path, mode);
+  if (!stream) {
+    // TODO: better to ask the stream about `errc`
+    ec = std::make_error_code(std::io_errc::stream);
+    fs::remove(path);
+  }
+
   ec.clear();
-  return std::make_pair(path, handle);
+  // FIXME: opened handle for the file won't be closed
```

Comment:
`std::fstream` opens the file by itself, there is no way to provide an already opened native handle

So our handle needs to be closed before exiting the function